src/SqlVirtualMachine/SqlVirtualMachine/SqlVirtualMachine/Cmdlet/UpdateAzureSqlVM.cs

```diff
@@ -0,0 +1,225 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Commands.ResourceManager.Common.Tags;
+using Microsoft.Azure.Commands.SqlVirtualMachine.Common;
+using Microsoft.Azure.Commands.SqlVirtualMachine.SqlVirtualMachine.Model;
+using static Microsoft.Azure.Commands.SqlVirtualMachine.Common.ParameterSet;
+
+namespace Microsoft.Azure.Commands.SqlVirtualMachine.SqlVirtualMachine.Cmdlet
+{
+    /// <summary>
+    /// Defines Update-AzSqlVM cmdlet
+    /// </summary>
+    [Cmdlet(VerbsData.Update, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "SqlVM", DefaultParameterSetName = NameInputObject, ConfirmImpact = ConfirmImpact.Medium, SupportsShouldProcess = true), OutputType(typeof(AzureSqlVMModel))]
+    public class UpdateAzureSqlVM : AzureSqlVMUpsertCmdletBase
+    {
+        /// <summary>
+        /// Resource group name of the sql virtual machine
+        /// </summary>
+        [Parameter(Mandatory = true,
+            ParameterSetName = NameParameterList,
+            Position = 0,
+            HelpMessage = HelpMessages.ResourceGroupSqlVM)]
+        [Parameter(Mandatory = true,
+            ParameterSetName = NameInputObject,
+            Position = 0,
+            HelpMessage = HelpMessages.ResourceGroupSqlVM)]
+        [ResourceGroupCompleter]
+        public string ResourceGroupName { get; set; }
+
+        /// <summary>
+        /// Name of the sql virtual machine
+        /// </summary>
+        [Parameter(Mandatory = true,
+            ParameterSetName = NameParameterList,
+            Position = 1,
+            HelpMessage = HelpMessages.NameSqlVM)]
+        [Parameter(Mandatory = true,
+            ParameterSetName = NameInputObject,
+            Position = 1,
+            HelpMessage = HelpMessages.NameSqlVM)]
+        [Alias("Name")]
+        [ResourceNameCompleter("Microsoft.SqlVirtualMachine/SqlVirtualMachines", "ResourceGroupName")]
+        public string SqlVMName { get; set; }
+
+        /// <summary>
+        /// License type of the new sql virtual machine
+        /// </summary>
+        [Parameter(Mandatory = false,
+            ParameterSetName = NameParameterList,
+            HelpMessage = HelpMessages.LicenseTypeSqlVM)]
+        [Parameter(Mandatory = false,
+            ParameterSetName = ResourceIdParameterList,
+            HelpMessage = HelpMessages.LicenseTypeSqlVM)]
+        [ValidateNotNullOrEmpty]
+        public string LicenseType { get; set; }
```

Comment:
@rocchiele same comment about adding the `PSArgumentCompleter` attribute if this parameter has a closed set of values